### PR TITLE
Add support for toml-formatted variables

### PR DESCRIPTION
**Pull request by mraspberry**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -83,6 +83,16 @@ to one of the following:
 
 To learn more about the loader, see `here <https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation>`_
 
+TOML
+~~~~~~~~~~
+
+`TOML`_ TOML aims to be a minimal configuration file format that's easy to read due to obvious semantics. TOML is designed to map unambiguously to a hash table.
```
Comment (BeyondEvil) [inline README.rst]:
```suggestion
`TOML`_ aims to be a minimal configuration file format that's easy to read due to obvious semantics. TOML is designed to map unambiguously to a hash table.
```

Also, please add a proper link at the bottom of the file (compare with `YAML`). 👍

File: test_variables.py [hunk 2]
```diff
@@ -84,7 +87,7 @@ def test(variables):
 
 def test_invalid_format(testdir, file_format):
     testdir.makepyfile("def test(variables): pass")
-    result = run(testdir, file_format, ["invalid"], raw=True)
+    result = run(testdir, file_format, ["invalid="], raw=True)
```
Comment (BeyondEvil) [inline test_variables.py]: Out of curiosity, why was this change needed?

Comment (mraspberry) [reply test_variables.py]: > Out of curiosity, why was this change needed?

This was needed because of some strangeness with TOML. A file containing something that looks like a key but with no equals sign or newline returns an empty dict, not a ValueError. It happens in at least two different python toml implementations so i'm guessing it's part of the spec.
